v0/pipeline/steps/cd/k8s-istio-provider-spec.yaml

```diff
@@ -0,0 +1,25 @@
+title: IstioProviderSpec
+type: object
+required:
+  - gateways
+  - hosts
+  - routes
+  - destinations
+properties:
+  gateways:
+    type: array
+    items:
+      type: string
+  hosts:
+    type: array
+    items:
+      type: string
+  routes:
```

Comment:
we should add routes and destinations to a sort of base object and here call "allOf" 

something like


/schema/base
{
  "type": "object",
  "properties": {
    "routes": { ... },
    "destinations": { ... }
  }
}


{
  "allOf": [{ "$ref": "/schema/base" }],
  "properties": {
    "gateways": { ... }
  }
}